pytest_factoryboy/fixture.py

```diff
@@ -43,41 +41,45 @@ def __call__(self, request: SubRequest) -> Any:
         return self.function(request)
 
 
-class RegisterProtocol(Protocol):
-    """Protocol for ``register`` function called with ``factory_class``."""
-
-    def __call__(self, factory_class: F, _name: str | None = None, **kwargs: Any) -> F:
-        """``register`` function called with ``factory_class``."""
-
-
+# register(AuthorFactory, ...)
+#
+# @register
+# class AuthorFactory(factory.Factory): ...
 @overload
-def register(
-    factory_class: None = None,
-    _name: str | None = None,
-    **kwargs: Any,
-) -> RegisterProtocol:
+def register(factory_class: F, _name: str | None = None, **kwargs: Any) -> F:
     ...
 
 
+# @register(...)
+# class AuthorFactory(factory.Factory): ...
 @overload
-def register(factory_class: F, _name: str | None = None, **kwargs: Any) -> F:
+def register(*, _name: str | None = None, **kwargs: Any) -> Callable[[F], F]:
     ...
 
 
 def register(
     factory_class: F | None = None,
     _name: str | None = None,
+    *,
+    _caller_locals: dict[str, Any] | None = None,
     **kwargs: Any,
-) -> F | RegisterProtocol:
+) -> F | Callable[[F], F]:
     r"""Register fixtures for the factory class.
 
     :param factory_class: Factory class to register.
     :param _name: Name of the model fixture. By default, is lowercase-underscored model name.
+    :param _caller_locals: Dictionary where to inject the generated fixtures. Defaults to the caller's locals().
     :param \**kwargs: Optional keyword arguments that override factory attributes.
     """
+    if _caller_locals is None:
+        _caller_locals = get_caller_locals()
 
     if factory_class is None:
-        return functools.partial(register, _name=_name, **kwargs)
+
+        def register_(factory_class: F) -> F:
```

Comment:
NIP: Personally, I don't see any benefit from returning such function instead of `functools.partial`. The main drawback here is that you cannot pass `**kwargs` (or `factory_kwargs` in the future) to the second call, but if it will be documented well it won't be a problem.